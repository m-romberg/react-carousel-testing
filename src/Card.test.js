import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", function () {
  // this is a low-value test, but better than nothing
  render(<Card caption="caption" src="img.png" currNum={1} totalNum={3} />);
});
// end

it("has the correct alt text & src", function () {
  const { container, debug } = render(
    <Card caption="caption" src="img.png" currNum={1} totalNum={3} />
  );
  const img = container.querySelector("img");
  debug(img);

  expect(img.getAttribute("alt")).toEqual("caption");
  expect(img.getAttribute("src")).toContain("img.png");
  //FIXME: checking whole dom object....
  // expect(container.querySelector(".Card-small")).toContain("1 of 3");
});
// end

it("matches snapshot", function () {
  const { container } = render(
    <Card caption="caption" src="img.png" currNum={1} totalNum={3} />
    );
  expect(container).toMatchSnapshot();
});