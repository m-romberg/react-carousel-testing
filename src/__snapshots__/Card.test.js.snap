// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      caption
    </h4>
    <img
      alt="caption"
      class="Card-image"
      src="img.png"
    />
    <small
      class="Card-small"
    >
      Image 
      1
       of 
      3
      .
    </small>
  </div>
</div>
`;
